pkg/download/automation/automation.go

```diff
@@ -0,0 +1,111 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package automation
+
+import (
+	"encoding/json"
+	"github.com/dynatrace/dynatrace-configuration-as-code/internal/log"
+	"github.com/dynatrace/dynatrace-configuration-as-code/internal/maps"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/client/automation"
+	config "github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/coordinate"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/parameter"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/parameter/value"
+	"github.com/dynatrace/dynatrace-configuration-as-code/pkg/config/v2/template"
+	v2 "github.com/dynatrace/dynatrace-configuration-as-code/pkg/project/v2"
+)
+
+var automationTypesToResources = map[config.AutomationType]automation.ResourceType{
+	config.AutomationType{Resource: config.Workflow}:         automation.Workflows,
+	config.AutomationType{Resource: config.BusinessCalendar}: automation.BusinessCalendars,
+	config.AutomationType{Resource: config.SchedulingRule}:   automation.SchedulingRules,
+}
+
+// Downloader can be used to download automation resources/configs
+type Downloader struct {
+	client *automation.Client
+}
+
+// NewDownloader creates a new [Downloader] for automation resources/configs
+func NewDownloader(client *automation.Client) *Downloader {
+	return &Downloader{
+		client: client,
+	}
+}
+
+// Download downloads all automation resources for a given project
+// If automationTypes is given it will just download those types of automation resources
+func (d *Downloader) Download(projectName string, automationTypes ...config.AutomationType) (v2.ConfigsPerType, error) {
+
+	// if no specific automation types are given, take all
+	if len(automationTypes) == 0 {
+		automationTypes = maps.Keys(automationTypesToResources)
+	}
+
+	// download automation resources
+	configsPerType := make(v2.ConfigsPerType)
+	for _, at := range automationTypes {
+		resource, ok := automationTypesToResources[at]
+		if !ok {
+			log.Warn("No resource mapping for automation type %s found", at.Resource)
+			continue
+		}
+		// download all configs of resource
+		response, err := d.client.List(resource)
+		if err != nil {
+			return nil, err
+		}
+
+		// nothing to do if there were no configs
+		// continue with next resource
+		if len(response.Results) == 0 {
+			continue
+		}
+		log.Info("Downloaded %d objects for automation resource %s", len(response.Results), string(at.Resource))
+
+		// convert and collect all objects into a slice of configs.Config values to return
+		var configs []config.Config
+		for _, obj := range response.Results {
+			configId := obj.Id
+			var content string
+			if indentedData, err := json.MarshalIndent(json.RawMessage(obj.Data), "", "  "); err == nil {
```

Comment:
nit: should we pull that marshal call into an internal util? I think every downloader uses it, and they all define 2 space indent and would have to change together if this ever changes